File: C#-part-2/BinarySearch/BinarySearch.cs
/*
Problem 11. Binary search

Write a program that finds the index of given element in a sorted array of integers by using the Binary search algorithm.
 */
using System;

	class BinarySearch
	{
		public static void Main()
		{
	    int[] bSearch = { 7, 14, 2, 5, 7, 9, 15, 77, 7 };
        Array.Sort(bSearch);
        int element = int.Parse(Console.ReadLine());
        Console.WriteLine(Array.BinarySearch(bSearch, element));
		}
	}

File: JavaScript-Fundamentals/05.Arrays/07. Binary search/BinarySearch.js
/**Problem 7. Binary search
 Write a script that finds the index of given element in a
 sorted array of integers by using the binary search algorithm
 */
function BinarySearch(args) {
    var array = args[0].split("\n").map(Number),
        n = +array[0],
        target = +array[array.length - 1],
        minIndex = 0,
        maxIndex = array.length - 1,
        currentIndex,
        currentElement;

    array.shift();
    array.pop();

    while (minIndex <= maxIndex) {
        currentIndex = ((minIndex + maxIndex) / 2) | 0;

        currentElement = +array[currentIndex];

        if (target > currentElement) {
            minIndex = currentIndex + 1;
        }
        else if (target < currentElement) {
            maxIndex = currentIndex - 1;
        }
        else {
            return currentIndex;
        }
    }

    return -1;
}
